vmdb/lib/workers/worker_base.rb

```diff
@@ -150,10 +151,15 @@ def log_prefix
   #
 
   def find_worker_record
-    @worker = self.class.corresponding_model.find_by_guid(@cfg[:guid])
+    @worker = self.class.corresponding_model.includes(:miq_server).find_by_guid(@cfg[:guid])
     do_exit("Unable to find instance for worker GUID [#{@cfg[:guid]}].", 1) if @worker.nil?
   end
 
+  def find_server_record
+    find_worker_record unless @worker
```

Comment:
I think the right approach is probably to not have a dependency on the worker record at all...that way, there's no need to require the dependency in the methods.  Or, initialize the `@server` ivar in the `find_worker_method`.  Which, at the time, seemed like a strange side effect for that method.